site_scons/buildutils.py

```diff
@@ -34,6 +36,373 @@
     TPathLike = TypeVar("TPathLike", Path, str)
 
 
+class Option:
+    """Object corresponding to SCons configuration option"""
+
+    def __init__(self,
+            name: str,
+            description: str,
+            default: Union[bool, str],
+            choices: list=None):
+        self.name = name
+        self.description = Option._deblank(description)
+        self.default = default
+        self.choices = choices
+
+        self._wrapper = textwrap.TextWrapper(width=80)
+        self.set_wrapper_indent(4)
+
+    @property
+    def parameters(self) -> tuple:
+        """Parameters of the SCons configuration option"""
+        if self.choices is None:
+            return self.name, self.description, self.default
+        return self.name, self.description, self.default, self.choices
+
+    def to_scons(self, env: "SCEnvironment"=None) -> "SCVariables":
+        """Convert option to SCons variable"""
+        if isinstance(self.default, bool):
+            return self.parameters
+
+        if isinstance(self.default, str):
+            out = list(self.parameters)
+            if env is not None and "$" in out[2]:
+                out[2] = env.subst(out[2])
+            return tuple(out)
+
+        raise TypeError(f"Invalid defaults option with type '{type(self.default)}'")
+
+    @property
+    def wrapper(self):
+        """Line wrapper for text output"""
+        return self._wrapper
+
+    def set_wrapper_indent(self, indent: int=4):
+        """Update indent used for line wrapping"""
+        self._wrapper.initial_indent = indent * " "
+        self._wrapper.subsequent_indent = indent * " "
+
+    @staticmethod
+    def _deblank(string: str) -> str:
+        """Remove whitespace before and after line breaks"""
+        out = [s.strip() for s in string.split("\n")]
+        if not len(out[-1]):
+            out = out[:-1]
+        out = "\n".join(out)
+        return out
+
+    def _build_title(self, backticks: bool=True, indent: int=3) -> str:
+        """Build title describing option and defaults"""
+        # First line: "* option-name: [ 'choice1' | 'choice2' ]"
+
+        def decorate(key: str, tick: bool=True):
+            key = f"'{key}'" if tick else key
+            return f"``{key}``" if backticks else key
+
+        # format choices
+        if isinstance(self, PathOption):
+            choices = f"path/to/{self.name}"
+            choices = f"{decorate(choices, False)}"
+        elif isinstance(self.choices, (list, tuple)):
+            choices = list(self.choices)
+            for yes_no in ["n", "y"]:
```

Comment:
```suggestion
            for yes_no in ("y", "n"):
```
For yes-no in "no", "yes" 😂